tasks/integration.yml

```diff
@@ -46,22 +46,91 @@
   loop: "{{ datadog_integration|dict2items }}"
   when: item.value.action == "remove" and ansible_facts.os_family == "Windows"
 
-# Install integrations
+# Check for existing integration installs / versions
 
-- name: Install pinned version of integrations (Unix)
-  command: "{{ datadog_agent_binary_path }} integration install {{ third_party }} {{ item.key }}=={{ item.value.version }}"
+- name: "Check for existing install / version of integrations (Unix)"
+  command:
+    argv:
+      - "{{ datadog_agent_binary_path }}"
+      - "integration"
+      - "show"
+      - "{{ integration.key }}"
+      - "-q"
   become: yes
   become_user: "{{ integration_command_user }}"
-  vars:
-    third_party: "{% if 'third_party' in item.value and item.value.third_party | bool %}--third-party{% endif %}"
+  check_mode: no
+  changed_when: >-
+    integration_results.rc != 0 or integration_results.stdout != integration.value.version
+
+  # Ignore command failure if integration is not already installed
+  failed_when: no
+  register: "integration_existing"
+  loop: "{{ datadog_integration | dict2items }}"
+  loop_control:
+    loop_var: 'integration'
+  when:
+    - "integration.value.action | lower == 'install'"
+    - "ansible_facts.os_family != 'Windows'"
```

Comment:
I prefer to keep the platform checks as the first condition - while at this case it doesn't matter, there is a place in this file where it does matter and it's good to keep things consistent.

Also, please remove the two blank lines below.